### ISSUE TITLE
BUG: Out of order messages when creating 2 bill splits while offline

### ISSUE THREAD

**Issue by neil-marcellini** [Expensify/App#13586]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go offline
2. From the global create menu +, select Split bill
3. Go through the prompt and select TWO or MORE participants that you have never split a bill with, and submit the request
4. Verify that the group chat is created with those participants and you are navigated to it (the whole chat should be greyed out)
5. Verify that the group chat has a message Split $X with <user1> and <user2> (this should be greyed out)
6. Verify that individual chats with each participant are created (these chats should be greyed out)
7. Verify that individual chats have the IOU message and preview (these messages should be greyed out)
8. From the group chat, tap + > Split bill and send another split bill request
9. Verify that the modal is dismissed
10. Verify that another split message shows up in the group chat (this message should be greyed out)
11. Verify that each individual chat with and show the new request and the total has been updated (these messages should be greyed out)
12. Turn on your internet connection and verify that the chats and messages are still the same, but are no longer greyed out.

## Expected Result:

1. Messages in the split chats are always in order
2. After creating the second split while offline the chats in the LHN are grayed out

## Actual Result:

1. Messages in the split chats are not always in order
2. After creating the second split while offline the chats in the LHN were no longer grayed out

## Workaround:
Go back online and ignore the UI flaws.

## Platform:

All
- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** v1.2.39-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** N/A
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:**

https://user-images.githubusercontent.com/26260477/207669109-f5ebb3ff-e1f7-4fde-86ae-e8ba1c689217.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @neil-marcellini 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1670980004459869

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d7aa95ed8ddc61fb</li>
        <li>Upwork Job ID: 1605621931009056768</li>
        <li>Last Price Increase: 2022-12-21</li>
    </ul>
</details>

**Comment by neil-marcellini** [comment]
I put up a draft PR. I think another PR might have already fixed the issue where the chats appear out of order. I'll test that later and put the PR up for review if so.

**Comment by neil-marcellini** [comment]
Today I was working on my monthly issues so I didn't get to this.

**Comment by neil-marcellini** [comment]
Today I was working on my weekly issues so I didn't get to this.

**Comment by neil-marcellini** [comment]
Tomorrow I will test the PR to fix the grayed out state in the LHN and I'll put it up for review as a partial fix. I'm holding the out of order messages part on the issue to fix the replay effect.

**Comment by neil-marcellini** [comment]
The grayed out state fix is merged. This issue https://github.com/Expensify/App/issues/13310 is the same problem so I'm going to close this one and let the problem get solved in the issue eventually.

**Comment by neil-marcellini** [comment]
Actually, I should leave this open until the related PR is deployed.

**Comment by neil-marcellini** [comment]
Here's the PR, GH doesn't seem to automatically link it right now https://github.com/Expensify/App/pull/13602

**Comment by neil-marcellini** [comment]
Since this issue is locked the Bug Zero checklist didn't post, so here it is.


**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@neil-marcellini] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@neil-marcellini] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@neil-marcellini] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@johncschuster] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:

**Comment by johncschuster** [comment]
@mananjadhav can you apply for the Upwork job, here?

https://www.upwork.com/jobs/~01d7aa95ed8ddc61fb

**Comment by johncschuster** [comment]
This has been paid out! I'll work on the regression test shortly.

**Comment by neil-marcellini** [comment]
@johncschuster any update on this, it's been a month. Also because the regressions steps weren't updated we have this issue as well https://github.com/Expensify/App/issues/14560.

**Comment by Beamanator** [comment]
@johncschuster how did you comment on this last **_YEAR_** :trollface: 

https://user-images.githubusercontent.com/3885503/217236929-ca14321b-731e-4ab7-b25f-f81ffaed08d7.mov

**Comment by johncschuster** [comment]
Oy! Sorry guys. I've been a bit behind on GH notifs. Working on the regression steps now!

**Comment by johncschuster** [comment]
Oh! It looks like the QA steps from the linked PR are already included in the TestSuite, [here](https://expensify.testrail.io/index.php?/cases/view/1971182&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=229063).

**Comment by johncschuster** [comment]
# PR
![2023-02-07_11-59-37](https://user-images.githubusercontent.com/5579997/217327991-91462f4e-1a26-4393-81ac-3833c9a3cc32.png)


# TestRail
![2023-02-07_11-59-13](https://user-images.githubusercontent.com/5579997/217328020-2513b9bd-8c7f-484d-9dea-2f2d2cb46e10.png)

**Comment by neil-marcellini** [comment]
Ok cool. The [other issue](https://github.com/Expensify/App/issues/14560) was not following these test steps as far as I can tell. Maybe we can make them more clear, or maybe they are good as is? What do you think @kbecciv?

**Comment by neil-marcellini** [comment]
We could update step 8 like so to describe the expected behavior regarding participants being grayed out on the second pending split.

`Navigate to the created group chat and tap + > Split bill and send another split bill request (the participants with a pending split should be grayed out)`

**Comment by johncschuster** [comment]
Great suggestion! I've created an issue for that proposed change, [here.](https://github.com/Expensify/Expensify/issues/260938)

**Comment by neil-marcellini** [comment]
🚀

**Comment by johncschuster** [comment]
Are we good to close this one out?

**Comment by neil-marcellini** [comment]
It looks like you already completed the payments so I updated the issue title. Let's wait until the regression test is actually updated and this issue is closed https://github.com/Expensify/Expensify/issues/260938. @johncschuster please ping the #qa channel in Slack. I find that's usually required to get someone on the issue quickly.

**Comment by MelvinBot** [comment]
@johncschuster, @neil-marcellini Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by neil-marcellini** [comment]
The regression test issue is close now so this is done!
